python/lsst/resources/http.py

```diff
@@ -838,7 +838,16 @@ def size(self) -> int:
                 if size is None:
                     raise ValueError(f"Content-Range header for {self} did not include a total file size")
                 return size
-
+            elif resp.status_code == requests.codes.range_not_satisfiable:
+                # 416 Range Not Satisfiable, which can occur on a GET for a 0
+                # byte file since we asked for 1 byte Range which is longer
+                # than the file.
+                #
+                # Servers are supposed to include a Content-Range header in
```

Comment:
I will say in passing that we have in the past managed to get Google to fix their S3 server implementation.

What happens if you have a GCS signed URL? Same problem or does it really matter whether the URI was s3 or gs originally?

Reply:
Not sure -- I don't have a convenient way to generate signed GCS URLs.

The 416 thing isn't Google-specific and is standards-compliant behavior, btw, so this change isn't only necessary for Google compatibility.  I just failed to consider what would happen with 0-byte files when I did the initial implementation.

The part that is non-compliant is that it doesn't also return a Content-Range including the size (specified as a SHOULD in the HTTP RFC).  Unless something else really weird and non-compliant is happening that size will always be 0 so I don't think it's especially important.  

(If the server doesn't want to or can't deal with the Range at all it's supposed to return 200 with the full content instead, which will result in correct behavior since we go down the branch that looks at Content-Length.  Slightly rude to the server but probably mostly theoretical -- implementations of S3 will almost certainly support Range requests.)

It looks like the HttpResourceHandle also assumes it's at EOF and bails early without trying to look at Content-Range if it gets a 416, so I don't think it's a problem.